site/components/src/main/java/uk/nhs/digital/common/components/catalogue/ApiCatalogueHubComponent.java

```diff
@@ -18,44 +18,44 @@
 import uk.nhs.digital.common.components.catalogue.filters.Section;
 import uk.nhs.digital.common.components.info.ApiCatalogueHubComponentInfo;
 
-import java.util.*;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
 import java.util.stream.Collectors;
 
-@ParametersInfo(
-    type = ApiCatalogueHubComponentInfo.class
-)
+@ParametersInfo(type = ApiCatalogueHubComponentInfo.class)
 public class ApiCatalogueHubComponent extends EssentialsListComponent {
 
     private static final Logger log = LoggerFactory.getLogger(ApiCatalogueHubComponent.class);
 
+    private ApiCatalogueFilterManager apiCatalogueFilterManager;
+
+    public void setApiCatalogueFilterManager(ApiCatalogueFilterManager apiCatalogueFilterManager) {
+        this.apiCatalogueFilterManager = apiCatalogueFilterManager;
+    }
+
```

Comment:
### Part 1/2 (unintended shared state)

_The original comment was lost :(, I try again_ 

**Once this is set, it's set!** To avoid side effects, change this to have two constructors. 

```
ApiCatalogueHubComponent() {
  /* Used by the framework */
}

ApiCatalogueHubComponent(ApiCatalogueFilterManager apiCatalogueFilterManager) {
  this.testableApiCatalogueFilterManager = apiCatalogueFilterManager;
}
```

Reply:
As discussed, constructor method doesn't helped here, getting some errors. Since we changed the object creation as Part 2/2 approach and we are not using threading here, unintended shared stared issue will not occur here.